lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemFactory.java

```diff
@@ -120,12 +126,21 @@ public S3FileSystemFactory(OpenTelemetry openTelemetry, S3FileSystemConfig confi
                 config.getSseKmsKeyId(),
                 Optional.empty(),
                 config.getCannedAcl());
+
+        this.uploadExecutor = new ThreadPoolExecutor(
+                0,
+                config.getStreamingMaxUploadThreads(),
+                60L,
+                TimeUnit.SECONDS,
+                new SynchronousQueue<>(),
+                daemonThreadsNamed("s3-upload-%s"));
```

Comment:
You should almost always use one of the default executors. Configuring `ThreadPoolExecutor` usually has some odd behaviors you don't expect. For example, this configuration will create 60 concurrent tasks, and then start rejecting new tasks.

You should instead use a `Executors.newCachedThreadPool`, and wrapper it with a `BoundedExecutor`.